include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan.h

```diff
@@ -121,18 +117,17 @@ single_pass_scan_impl(sycl::queue __queue, _InRange&& __in_rng, _OutRange&& __ou
     static_assert(_Inclusive, "Single-pass scan only available for inclusive scan");
 
     const ::std::size_t n = __in_rng.size();
-#ifdef _ONEDPL_SCAN_ITER_SIZE
-    constexpr ::std::size_t __elems_per_workitem = _ONEDPL_SCAN_ITER_SIZE;
-#else
-    constexpr ::std::size_t __elems_per_workitem = 8;
-#endif
-    // Next power of 2 greater than or equal to __n
-    auto __n_uniform = n;
-    if ((__n_uniform & (__n_uniform - 1)) != 0)
-        __n_uniform = oneapi::dpl::__internal::__dpl_bit_floor(n) << 1;
-    std::size_t num_workitems = __n_uniform / __elems_per_workitem;
-    std::size_t wgsize = num_workitems > 256 ? 256 : num_workitems;
-    std::size_t num_wgs = oneapi::dpl::__internal::__dpl_ceiling_div(num_workitems, wgsize);
+
+    constexpr ::std::size_t wgsize = _KernelParam::__workgroup_size;
+    constexpr ::std::size_t elems_per_workitem = _KernelParam::__elems_per_workitem;
+
+    // Avoid non_uniform n by padding up to a multiple of wgsize
+    ::std::size_t num_workitems = n / elems_per_workitem;
+    num_workitems = num_workitems + wgsize - 1;
+    num_workitems -= (num_workitems % wgsize);
+
+    ::std::size_t num_wgs = num_workitems / wgsize;
+    std::uint32_t elems_in_tile = wgsize * elems_per_workitem;
```

Comment:
I think 'the oneDPL way' of doing this is to use their `ceiling_div` helper fn. e.g. (from parallel_backend_sycl_reduce.h):

```c++
    constexpr _Size __size_per_work_group = __iters_per_work_item_device_kernel * __work_group_size;
    const _Size __n_groups = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __size_per_work_group);
```